src/environment.ts

```diff
@@ -205,7 +207,8 @@ export class SecureEnvironment {
     getSecureArray(a: RawArray): SecureArray {
         // identity of the new array correspond to the inner realm
         const SecureArray = (this.realm.global as any).Array as ArrayConstructor;
-        return new SecureArray(...a).map((raw: RawValue) => this.getSecureValue(raw));
+        const b: SecureValue[] = map(a, (raw: RawValue) => this.getSecureValue(raw));
```

Comment:
@jdalton just getting a brand new array via map for all secure values, then using `...b` to let the grammar to build a new array with the identity of the sandbox, and the proper values.